### PR TITLE
#6838: Add tracy timeout for op reprots

### PR DESCRIPTION
**Pull request by mo-tenstorrent**
Added 15s timeout on tracy capture. Capture actually quickly returns at that stage of the script but long timeout was added just in case. 

This avoids hanging python scripts that are waiting on any tracy data.

### CODE REVIEW

**Review by TT-billteng — APPROVED** [review]

**Comment by mo-tenstorrent** [comment]
https://github.com/tenstorrent-metal/tt-metal/actions/runs/8458778645 Post-commit

File: tt_metal/tracy.mk [hunk 1]
```diff
@@ -2,6 +2,8 @@ TRACY_LIB = $(LIBDIR)/libtracy.so
 TRACY_INCLUDES = -I$(TT_METAL_HOME)/tt_metal/third_party/tracy/public/tracy/
 TRACY_LDFLAGS = $(LDFLAGS)
 TRACY_DEFINES = -DTRACY_NO_CONTEXT_SWITCH
+TRACY_BUILD_DIR = $(TT_METAL_HOME)/build/profiler/bin
```
Comment (tt-rkim) [inline tt_metal/tracy.mk:5]: In CMake, would this be `ExternalProject_Add` with a custom `BUILD_CMD` ? @mo-tenstorrent @TT-billteng

Comment (TT-billteng) [reply tt_metal/tracy.mk:5]: yes I think so

File: module.mk [hunk 2]
```diff
@@ -132,7 +134,6 @@ LIBS_TO_BUILD += \
 endif
 
 LIBS_TO_BUILD += \
-	tracy \
```
Comment (TT-billteng) [inline module.mk:135]: is this target obselete now?

**Review by TT-billteng — APPROVED** [review]
nice!

**Review by tt-rkim — APPROVED** [review]

**Comment by mo-tenstorrent** [comment]
https://github.com/tenstorrent-metal/tt-metal/actions/runs/8543057823/job/23405998138 Device perf is only failing due to know regression on ttnn/resnet

https://github.com/tenstorrent-metal/tt-metal/actions/runs/8543077747 post commit passing

**Comment by mo-tenstorrent** [comment]
https://github.com/tenstorrent-metal/tt-metal/actions/runs/8543948789 Profiler pass after minor changes

**Pull request merged**